frontend/src/lib/hooks/useAddressBook/index.ts

```diff
@@ -0,0 +1,19 @@
+import {
+  PersistentStateKey,
+  usePersistentState,
+} from '../../../state/persistentState';
+
+export const useAddressBook = () => {
+  const [addresses, setAddresses] = usePersistentState<string[]>(
+    PersistentStateKey.AddressBook,
+    []
+  );
+
+  return {
+    addresses,
+    addAddress: (newAddress: string) =>
```

Comment:
The issue with those setters is that compared to `setAddress` is that when they mentioned in hook deps they would cause those hooks to "re-run". Usually it's not an issue, but in some complex component might add some headache.

Also, when we stop using useState "traditional" return type of `[value, setValue]` we lose ability to wrap them with reusable hooks such as `useStateCorrector` or `usePresentState` which expect the traditional `[value, setValue]` format.